File: androidlib/src/main/java/com/laka/androidlib/widget/NoScrollViewPager.java
package com.laka.androidlib.widget;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;

/**
 * @Author:Rayman
 * @Date:2018/4/24
 * @description:修改滑动逻辑处理
 **/
public class NoScrollViewPager extends ViewPager {

    private boolean smoothScroll = false;
    private boolean isCanScroll = false;

    public NoScrollViewPager(Context context) {
        super(context);
    }

    public NoScrollViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void setCanScroll(boolean isCanScroll) {
        this.isCanScroll = isCanScroll;
    }

    @Override
    public void scrollTo(int x, int y) {
        super.scrollTo(x, y);
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {

        return isCanScroll && super.onTouchEvent(ev);
    }

    @Override
    public void setCurrentItem(int item, boolean smoothScroll) {
        super.setCurrentItem(item, smoothScroll);
    }

    @Override
    public void setCurrentItem(int item) {
        // true, 需要切换动画。false, 不需要
        super.setCurrentItem(item, smoothScroll);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return isCanScroll && super.onInterceptTouchEvent(ev);
    }

    @Override
    protected boolean canScroll(View v, boolean checkV, int dx, int x, int y) {
        return super.canScroll(v, checkV, dx, x, y);
    }

    public NoScrollViewPager setSmoothScroll(boolean smoothScroll) {
        this.smoothScroll = smoothScroll;
        return this;
    }
}

File: Main/src/main/java/com/laka/ergou/common/widget/NoScrollViewPager.kt
package com.laka.ergou.common.widget

import android.content.Context
import android.support.v4.view.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent


/**
 * @Author:summer
 * @Date:2019/3/7
 * @Description:
 */
class NoScrollViewPager : ViewPager {

    private var mEnableScroll = false

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)

    fun setEnableScroll(enable: Boolean) {
        this.mEnableScroll = enable
    }

    override fun onTouchEvent(arg0: MotionEvent): Boolean {
        return if (!mEnableScroll)
            false
        else
            super.onTouchEvent(arg0)
    }

    override fun onInterceptTouchEvent(arg0: MotionEvent): Boolean {
        return if (!mEnableScroll)
            false
        else
            super.onInterceptTouchEvent(arg0)
    }


}